spotify-api/spotifyRequests.ts

```diff
@@ -206,3 +206,216 @@ export const fetchArtistImagesBySpotifyId = async (
       throw err;
     });
 };
+
+export interface spotifyYoutubeTrackMapping {
+  youtube: {
+    name: string;
+  };
+  spotify?: {
+    name: string;
+    uri: string;
+  };
+}
+const fetchSingleSpotifyTrackByYoutubePlaylistURL = (
+  accessToken: string,
+  playlistTitle: { title: string },
+  maxRetries: number = 3,
+  currentRetry: number = 0
+): Promise<spotifyYoutubeTrackMapping> => {
+  return axios
+    .get<SpotifyApi.SearchResponse>(
+      queryParamsStringify('https://api.spotify.com/v1/search', {
+        q: playlistTitle.title,
+        type: 'track',
+        limit: '1',
+      }),
+      {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      }
+    )
+    .then((response) => {
+      // item limit is 1 above
+      const SpotifyTrackInfo: spotifyYoutubeTrackMapping | undefined = response.data.tracks?.items.map((track) => {
+        return {
+          youtube: {
+            name: playlistTitle.title,
+          },
+          spotify: {
+            uri: track.uri,
+            name: track.name,
+          },
+        };
+      })[0];
+
+      if (SpotifyTrackInfo === undefined) {
+        throw new Error('No spotify track found for video with name ' + playlistTitle.title);
+      }
+
+      return SpotifyTrackInfo;
+    })
+    .catch((err: AxiosError) => {
+      if (err.response?.status === HttpStatusCode.TooManyRequests) {
+        if (currentRetry < maxRetries) {
+          // Retry the request after a delay
+          const retryDelay = parseInt(err.response!.headers['retry-after'] as string) * 1000;
```

Comment:
rename to `retryDelayMs`, when working with time, it's good to suffix with the time unit